sstables/compaction.cc

```diff
@@ -965,7 +965,7 @@ class regular_compaction : public compaction {
             std::for_each(exhausted, _sstables.end(), [this] (shared_sstable& sst) {
                 _compacting_for_max_purgeable_func.erase(sst);
                 // Fully expired sstable is not actually compacted, therefore it's not present in the compacting set.
-                _compacting->erase(sst);
+                _compacting->erase(sst, false);
```

Comment:
This change was needed because we're dealing with an exceptional case here. This is the only place where we're modifying an sstable_set even though there might be some sstable readers that are using it. The false signals just that - modify the last version even if it has already been captured.